setools/terulequery.py

```diff
@@ -162,13 +165,21 @@ def results(self):
             #
             # Matching on target type
             #
-            if self.target and not match_indirect_regex(
+            if self.target and not self.target_self and not \
```

Comment:
There should also be a check after the debug messages above that raises and exception if `self.target` and `self.target_self` are set, as they conflict. That will simplify logic down here.

There should be a warning level log message for using target_regex or target_indirect with target_self since the former are ignored with self.